inspirehep/modules/references/processors.py

```diff
@@ -193,62 +199,62 @@ def set_pubnote(self, pubnote):
+            self.obj['reference']['arxiv_eprints'].append(_normalize_arxiv(repno))
         else:
-            self._ensure_field('publication_info', {})
-            self.obj['publication_info']['reportnumber'] = repno
+            self._ensure_reference_field('publication_info', {})
+            self.obj['reference']['publication_info']['reportnumber'] = repno
 
     def add_uid(self, uid):
         """Add unique identifier in correct field."""
         # We might add None values from wherever. Kill them here.
         uid = uid or ''
         if _is_arxiv(uid):
-            self._ensure_field('arxiv_eprints', [])
-            self.obj['arxiv_eprints'].append(_normalize_arxiv(uid))
+            self._ensure_reference_field('arxiv_eprints', [])
+            self.obj['reference']['arxiv_eprints'].append(_normalize_arxiv(uid))
         elif idutils.is_doi(uid):
-            self._ensure_field('dois', [])
-            self.obj['dois'].append(idutils.normalize_doi(uid))
+            self._ensure_reference_field('dois', [])
+            self.obj['reference']['dois'].append(idutils.normalize_doi(uid))
         elif idutils.is_handle(uid):
-            self._ensure_field('persistent_identifiers', [])
+            self._ensure_reference_field('persistent_identifiers', [])
             value = idutils.normalize_handle(uid)
             if not value.startswith('hdl:'):
                 # Prone to the day in which normalize_handle might prepend
                 # 'hdl:'.
                 value = u'hdl:{}'.format(value)
-            self.obj['persistent_identifiers'].append(value)
+            self.obj['reference']['persistent_identifiers'].append(value)
         elif self.RE_VALID_CNUM.match(uid):
-            self._ensure_field('publication_info', {})
-            self.obj['publication_info']['cnum'] = uid
+            self._ensure_reference_field('publication_info', {})
+            self.obj['reference']['publication_info']['cnum'] = uid
         else:
             # idutils.is_isbn has a different implementation than normalize
             # isbn. Better to do it like this.
             try:
                 isbn = idutils.normalize_isbn(uid)
-                self._ensure_field('publication_info', {})
-                self.obj['publication_info']['isbn'] = isbn
+                self._ensure_reference_field('publication_info', {})
+                self.obj['reference']['publication_info']['isbn'] = isbn
             # See https://github.com/nekobcn/isbnid/issues/2 and
             # https://github.com/nekobcn/isbnid/issues/3 for understanding the
             # long exception list.
             except (ISBNError, ISBNRangeError, UnicodeEncodeError):
                 pass
 
     def add_collaboration(self, collaboration):
-        self._ensure_field('collaboration', [])
-        self.obj['collaboration'].append(collaboration)
+        self._ensure_reference_field('collaboration', [])
+        self.obj['reference']['collaboration'].append(collaboration)
```

Comment:
If in the `_ensure_reference_field` function you return the object at the end, you could do for all these:

```
ref_colaboration = self._ensure_reference_field('collaboration', [])
ref_collaboration.append(collaboration)
```
Or even (very minimalistic and a bit hard to read)
```
self._ensure_reference_field('collaboration', []).append(collaboration)
```